src/app/users/user-profile-update/user-profile-update.component.ts

```diff
@@ -0,0 +1,124 @@
+import { Component, OnInit } from '@angular/core';
+import { UserService } from '../../shared/user.service';
+import { Location } from '@angular/common';
+import {
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  FormArray,
+  Validators
+} from '@angular/forms';
+import { CouchService } from '../../shared/couchdb.service';
+import { Observable } from 'rxjs/Observable';
+import { of } from 'rxjs/observable/of';
+import { MatFormField, MatFormFieldControl } from '@angular/material';
+import { ActivatedRoute, ParamMap } from '@angular/router';
+
+@Component({
+  templateUrl: './user-profile-update.component.html'
+})
+export class UserProfileUpdateComponent implements OnInit {
+  user: any;
+  educationLevel = [ '1', '2', '3', '4', '5', '6' , '7', '8', '9', '11', '12', 'Higher' ];
+  readonly dbName = '_users'; // make database name a constant
+  editForm: FormGroup;
+  file: any;
+
+  constructor(
+    private location: Location,
+    private fb: FormBuilder,
+    private couchService: CouchService,
+    private userService: UserService,
+    private route: ActivatedRoute
+  ) {
+    this.userData();
+  }
+
+  ngOnInit() {
+    Object.assign(this, this.userService.get());
+  }
+
+  userData() {
+    this.editForm = this.fb.group({
+      firstName: [ '', Validators.required ],
+      middleName: '',
+      lastName: [ '', Validators.required ],
+      login: [ { value: '', disabled: true }, Validators.required ],
+      email: [ '', [ Validators.required,  Validators.pattern ('[^ @]*@[^ @]*') ] ],
+      language: [ '', Validators.required ],
+      phoneNumber: [ '', Validators.required ],
+      birthDate: [ '', Validators.required ],
+      gender: [ '', Validators.required ],
+      level: [ '', Validators.required ]
+    });
+    this.editForm.setValue({firstName: '', middleName: '', lastName: '', login: '', email: '', language: ''
```

Comment:
This is not required.  You are already setting the value of all of the form fields to blank in `this.fb.group({...`

Reply:
@paul we have made separate issue for change password so that issue #272 can be merged